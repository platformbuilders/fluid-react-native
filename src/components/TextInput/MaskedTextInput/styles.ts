import styled from 'styled-components/native';
import { TextInputMask } from '@platformbuilders/react-native-masked-text';
import { TextInput as TextInputStyle } from '../styles';

const Input = TextInputStyle.withComponent(TextInputMask);

type Props = {
  placeholderTextColor?: string;
};

export const TextInput = styled(Input).attrs((props: Props) => ({
  placeholderTextColor: props.placeholderTextColor
    ? props.placeholderTextColor
    : '#72727260',
}))`
  max-width: 100%;
`;
