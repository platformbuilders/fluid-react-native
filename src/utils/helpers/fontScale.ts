import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
