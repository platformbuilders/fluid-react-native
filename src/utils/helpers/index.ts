export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
