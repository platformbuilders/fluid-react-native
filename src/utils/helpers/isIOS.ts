import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
