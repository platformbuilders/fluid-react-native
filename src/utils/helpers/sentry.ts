import { Sentry } from '.';

/**
 * Generic function to be used with Sentry. Under the hood, it's called the `addBreadcrumb` method
 * from Sentry API, which adds logs about user's actions prior to an error or crash.
 *
 * Read [this research doc](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/89ba7171-90e3-4e5b-9856-fe36be0bed14/Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20230119%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20230119T180359Z&X-Amz-Expires=86400&X-Amz-Signature=b0395d3285cebaa0562ac64b30ed50a3803928181b7a40428c81dfd000442f03&X-Amz-SignedHeaders=host&response-content-disposition=filename%3D%22Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf%22&x-id=GetObject)
 * to get more information about it's usage.
 * @param param0
 * @returns
 */
export const handleSentryTraceability = ({
  useSentryTraceability = false,
  category,
  data,
  event_id,
  level = 'info',
  message,
  timestamp,
  type,
}: HandleSentryTraceabilityProps) => {
  if (!useSentryTraceability) return;

  Sentry.addBreadcrumb({
    category,
    data,
    event_id,
    level,
    message,
    timestamp,
    type,
  });
};
