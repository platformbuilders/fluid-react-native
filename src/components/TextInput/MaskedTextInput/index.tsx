import React, { FC } from 'react';
import { MaskedTextInputType, InputStatus } from '../../../types';
import { TextInput } from './styles';

const MaskedTextInput: FC<MaskedTextInputType> = ({
  maskType,
  inputRef,
  id,
  accessibility,
  contrast = false,
  multiline = false,
  status = InputStatus.Default,
  ...props
}) => (
  <TextInput
    {...props}
    id={id || accessibility}
    status={status}
    accessibility={accessibility}
    testID={id || accessibility}
    accessibilityLabel={accessibility}
    ref={inputRef}
    contrast={contrast}
    multiline={multiline}
    type={maskType}
    underlineColorAndroid="transparent"
  />
);

export default MaskedTextInput;
