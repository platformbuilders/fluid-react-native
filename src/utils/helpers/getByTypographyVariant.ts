import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
