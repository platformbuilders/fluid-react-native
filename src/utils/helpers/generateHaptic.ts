import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
