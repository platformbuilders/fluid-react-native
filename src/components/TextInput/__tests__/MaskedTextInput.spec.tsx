import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import MaskedTextInput from '../MaskedTextInput';
import { theme } from '../../../test/helpers';

describe('<MaskedTextInput />', () => {
  it('should render textinput', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <MaskedTextInput
          id="test"
          variant="headline"
          accessibility=""
          maskType="cpf"
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
