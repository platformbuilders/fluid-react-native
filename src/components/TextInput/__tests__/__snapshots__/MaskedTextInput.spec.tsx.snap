// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaskedTextInput /> should render textinput 1`] = `
<TextInput
  accessibility=""
  accessibilityLabel=""
  allowFontScaling={true}
  contrast={false}
  id="test"
  keyboardType="numeric"
  multiline={false}
  onChangeText={[Function]}
  placeholderTextColor="#72727260"
  rejectResponderTermination={true}
  status="DEFAULT"
  style={
    Array [
      Object {
        "borderWidth": 0,
        "color": "#212121",
        "flexGrow": 1,
        "fontSize": 33,
        "lineHeight": 43,
        "marginTop": 0,
        "minHeight": 37.714285714285715,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
  testID="test"
  textAlign="left"
  type="cpf"
  underlineColorAndroid="transparent"
  value=""
  variant="headline"
/>
`;
